### ISSUE TITLE
Fetch Navigation and breadcrumb data

### ISSUE THREAD

**Issue by datakurre** [collective/gatsby-source-plone#71]

**Comment by datakurre** [comment]
These seem to be “expansions” on Plone REST API.

Maybe there could be an option list on which expansions should be fetched with data. By default it should have navigation and breadcrumbs enabled.

**Comment by datakurre** [comment]
https://plonerestapi.readthedocs.io/en/latest/expansion.html

**Comment by ajayns** [comment]
> These seem to be “expansions” on Plone REST API.

I was unaware of this, checking it out, it looks pretty helpful. 


> By default it should have navigation and breadcrumbs enabled.

Meaning by default, we should have the expanded the `breadcrumbs` and `navigation` data?

**Comment by datakurre** [comment]
Yes.

**Comment by datakurre** [comment]
Helpful indeed. From the docs

```
Here is an example of a request that expands all possible expansions:

GET /plone/front-page?expand=actions,breadcrumbs,navigation,schema,workflow HTTP/1.1
```
So you only need to make ``actions,breadcrumbs,navigation,schema,workflow`` a configurable list in gatsby-config.js, defaulting to breadcrumbs and navigation if the option is not present. After that the data should be available in graphql :-)

**Comment by ajayns** [comment]
Oh I see. So the things I'll need to is add options for fetching each of them and then accordingly fetchData for each content object. Hmm I'll first explore the API a bit and then come up with a solid approach, I think I'll have to modify the `fetchData` helper function quite a bit.

**Comment by ajayns** [comment]
@datakurre What would you think would be a good way to select these expansions? A single array in options called expansions where users can input an array of expansions or individual options for each expansion?

**Comment by datakurre** [comment]
@ajayns Single array of strings.

**Comment by datakurre** [comment]
I assume that expansions are implemented so that users might have implemented custom expansions.

**Comment by datakurre** [comment]
Fixed by #77
